### DISCUSSION TITLE
Migrate all monitoring statistics to a new ESP32

### DISCUSSION THREAD

**Discussion by dml33** [OpenEVSE/openevse_esp32_firmware#697] [Q&A]
Hi.

I want to change my current ESP32 for another one with an external antenna.

I would like to place all the monitoring information from the old ESP32 to the new ESP32.

How can I do it?

I see that in the `Developer` section, if I click on `Export all data`, I can get a json file with all the statistics I would like to save into the new ESP32, but I do not know how to import them...

Thanks!

**Comment by glynhudson** [comment]
You should be able to export and import a config backup via the Firmware page: 
 
![Screenshot from 2023-07-19 00-29-44](https://github.com/OpenEVSE/ESP32_WiFi_V4.x/assets/758844/08f61537-dc06-427a-8f14-dd8cf0e09e68)

**Reply by dml33** [reply]
Thanks for the reply.

But, in that Firmware page, only the settings, in a `config.json` file, can be downloaded and uploaded again.

I am looking for the monitoring statistics, such as:

![image](https://github.com/OpenEVSE/ESP32_WiFi_V4.x/assets/25033130/137de5a4-7fd4-442a-bcff-23d1d6b6f9c9)

These statistics can be exported to a `export.json` file in the Developer page, together with much more information, including the above-mentioned settings, via the `Export all data` button:

![image](https://github.com/OpenEVSE/ESP32_WiFi_V4.x/assets/25033130/4d720767-16cc-4d36-ac69-00c50084cc52)

However, this `export.json` file cannot be uploaded/imported to another ESP32, just to continue having your Energy meter numbers per week, month or year, the relay total toggles, etc.

Is there any way to do so? Thanks!

**Reply by dml33** [reply]
@glynhudson Then, can the `export.json` file be imported somehow into another ESP32?

Thanks!

**Comment by KipK** [comment]
No there's no way yet to import previous data logs.

**Reply by dml33** [reply]
Thanks for the reply.

Would it be possible in a near future? It would be quite nice in case of an ESP32 failure or  a migration to another one. 

I my case, for example, I am migrating to an ESP32 with an external antenna, so I could remove a WiFi extender, which is making me crazy.

I would like to keep my historical data to have a register of consumptions/costs and to know how much I have saved with the EV :)))

Thanks!
